Modules/CalcOffence-3_0.lua

```diff
@@ -1599,6 +1602,20 @@ function calcs.offence(env, actor, activeSkill)
 		else
 			output.ChillChanceOnCrit = 100
 		end
+		if skillModList:Flag(cfg, "CritAlwaysAltAilments") and not skillModList:Flag(cfg, "NeverCrit") and skillFlags.hit then
+			output.ScorchChanceOnCrit = 100
+			output.BrittleChanceOnCrit = 100
+			output.SapChanceOnCrit = 100
+		else
+			output.ScorchChanceOnCrit = 0
+			output.BrittleChanceOnCrit = 0
+			output.SapChanceOnCrit = 0
+			if skillModList:Flag(cfg, "CritAlwaysAltAilments") and not skillModList:Flag(cfg, "NeverCrit") then
```

Comment:
This code block has to be like this, else the flags won't be correctly set on hitting skills:
```lua
if skillModList:Flag(cfg, "CritAlwaysAltAilments") and not skillModList:Flag(cfg, "NeverCrit") then
    skillFlags.inflictScorch = true
    skillFlags.inflictBrittle = true
    skillFlags.inflictSap = true
end
if skillModList:Flag(cfg, "CritAlwaysAltAilments") and not skillModList:Flag(cfg, "NeverCrit") and skillFlags.hit then
    output.ScorchChanceOnCrit = 100
    output.BrittleChanceOnCrit = 100
    output.SapChanceOnCrit = 100
else
    output.ScorchChanceOnCrit = 0
    output.BrittleChanceOnCrit = 0
    output.SapChanceOnCrit = 0
end
```